packages/core/lib/rpc-util.js

```diff
@@ -14,36 +45,24 @@ export function createRpcService(service) {
     `invalid routes: ${rpcMethods} for service ${service.constructor.name}`,
   );
 
-  return rpcMethods.map(methodResolver => {
-    const routeName = methodResolver.name;
-    const methodName = `${name}.${routeName}`;
+  let methods = rpcMethods;
 
-    return {
-      name: methodName,
-      resolver: async (params = {}) => {
-        try {
-          let result;
+  if (!Array.isArray(rpcMethods)) {
+    return Object.keys(rpcMethods).map(key => {
+      return createMethodResolver({
+        methodFn: rpcMethods[key],
+        methodName: key,
+        service,
+      });
+    });
+  }
 
-          if (!methodResolver) {
-            throw new Error('Resolver is undefined');
-          }
-
-          if (params.__args) {
-            result = methodResolver.apply(service, params.__args);
-          } else {
-            result = methodResolver.apply(service, [params]);
-          }
-
-          return Promise.resolve(result).then(value => {
-            Logger.debug(`Result for ${methodName}`, value);
-            return value;
-          });
-        } catch (err) {
-          Logger.debug(`Error for ${methodName}`, err.toString());
-          throw err;
-        }
-      },
-    };
+  return methods.map(methodFn => {
+    return createMethodResolver({
+      methodFn,
+      methodName: methodFn.name,
```

Comment:
This method resolution works in the webview services. However the react native service, which will handle the storage, doesn't return the proper name.

Reply:
I'm keeping two ways to define json rpc services. Ideally we should refactor all the services later, just to avoid confusion and potential bugs